torch/_dynamo/output_graph.py

```diff
@@ -1366,9 +1370,30 @@ def add_output_instructions(self, prefix: List[Instruction]) -> None:
         self.output_instructions.extend(prefix)
         self.should_exit = True
 
-    def install_global(self, name, value) -> None:
+    def install_global_unsafe(self, name, value) -> None:
+        """
+        WARNING: prefer the safer `install_global_once`.
+        torch.compile instances should be independent of each other;
+        one footgun is to have one instance depend on the existence of
+        a global installed by another instance. This can happen if we mangle
+        a global the same way across both instances.
+        """
         self.cleanups.append(CleanupHook.create(self.global_scope, name, value))
 
+    def install_global_once(self, name, value) -> str:
+        """
+        Install a global once per frame being compiled. A frame graph breaking
+        into e.g. 2 graphs counts as two for the purposes of this function.
+
+        Returns the name of the newly installed global.
+        """
+        name = f"{name}_c{self.compile_id}"
```

Comment:
why do you need compile_id here?

Reply:
We are trying to prevent global sharing between different instances of torch.compile. I'm using "different output graphs" as the proxy for "different instances of torch.compile"

Reply:
The problem we ran into with test flakiness is that one output_graph would install `___module_torch_12345678`, another output_graph would rely on it, and then the first output_graph would die, taking `___module_torch_12345678` with it. Instead, each output_graph should have its own `___module_torch_12345678`, which is what compile_id is for.